layers/+emacs/outshine/packages.el

```diff
@@ -0,0 +1,65 @@
+;;; packages.el --- outshine layer packages file for Spacemacs.
+;;
+;; Copyright (c) 2012-2019 Sylvain Benner & Contributors
+;;
+;; Author: Langston Barrett <langston.barrett@gmail.com>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+;;; Commentary:
+
+;; Briefly, each package to be installed or configured by this layer should be
+;; added to `outshine-packages'. Then, for each package PACKAGE:
+;;
+;; - If PACKAGE is not referenced by any other Spacemacs layer, define a
+;;   function `outshine/init-PACKAGE' to load and initialize the package.
+
+;; - Otherwise, PACKAGE is already referenced by another Spacemacs layer, so
+;;   define the functions `outshine/pre-init-PACKAGE' and/or
+;;   `outshine/post-init-PACKAGE' to customize the package as it is loaded.
+
+;;; Code:
+
+(defconst outshine-packages
+  '(outshine
+    outorg))
+
+(defun outshine/init-outshine ()
+  (use-package outshine
+    :defer t
+    :init
+    (progn
+      (add-hook 'prog-mode-hook          'outline-minor-mode)
+      (add-hook 'outline-minor-mode-hook 'outshine-mode))
+    :config
+    (progn
+       (spacemacs/declare-prefix "aoo" "outline")
+       (spacemacs/set-leader-keys
+         "aooS"  'outline-show-all
```

Comment:
These bindings seem quite unwieldy - I don't think there's any other layer in Spacemacs which defines such long sequences of keys, the maximum I've seen is 3 characters long. 

In my local outline-related config I have these bound to the `SPC n` prefix for outli[n]e, I wonder if that's a reasonable default to have? 

```elisp
(spacemacs/set-leader-keys
  ;; visibility
  "nA" 'outline-show-all
  "nB" 'outline-show-branches
  "nS" 'outline-show-subtree
  "n TAB" 'outshine-cycle-buffer

  ;; Structural edits
  "nJ" 'outline-move-subtree-down
  "nK" 'outline-move-subtree-up
  "n<" 'outline-promote ; C-u for single heading
  "n>" 'outline-demote
  "nv" 'outline-mark-subtree
  "n RET" 'outshine-insert-heading

  ;; Outorg/Navi commands
  "ne" 'outorg-edit-as-org
  "nT" 'outshine-todo
  "nN" 'outshine-navi

  ;; Narrowing
  "nn" 'outshine-narrow-to-subtree
  "nw" 'narrow-or-widen-dwim
  "nW" 'widen
)
```

(my outline navigation commands are in a separate hydra with entry points "gj" and "gk", which may be even more of a personal preference)

Reply:
`SPC n` is bound by default. I couldn't think of an appropriate prefix which is shorter, though I'd be happy to find one. `SPC a o` is already used for some org-related stuff, so I thought it seemed appropriate. 

For navigation, a transient state as you suggest might be nice.

Reply:
I repeatedly find myself using the org keybindings on outorg headlines from muscle memory, only to receive a message like `<C-return> is undefined` when I want to create a new headline.

Also, I think the `SPC a` prefix is more appropriate for "applications" embedded in emacs, rather than modes.
I would argue that activating outshine/outorg is a conscious decision, thus justifying more "direct" bindings for its usage.

Would it be possible/feasible to provide the default org headline editing commands on the minor mode map, somehow with a low priority as not to disturb any major mode bindings?